charts/ColorLegendView.tsx

```diff
@@ -0,0 +1,247 @@
+import * as React from "react"
+import { computed, action } from "mobx"
+import { observer } from "mobx-react"
+
+import { Bounds } from "./Bounds"
+import { getRelativeMouse, sortBy } from "./Util"
+import { ColorScaleBin, CategoricalBin } from "./ColorScaleBin"
+import {
+    ColorLegend,
+    NumericColorLegend,
+    CategoricalColorLegend
+} from "./ColorLegend"
+
+const FOCUS_BORDER_COLOR = "#111"
+
+export interface ColorLegendViewProps {
+    legend: ColorLegend
+    onMouseOver: (d: ColorScaleBin) => void
+    onMouseLeave: () => void
+}
+
+@observer
+export class ColorLegendView extends React.Component<ColorLegendViewProps> {
```

Comment:
Question: could you have both a numeric and categorical legend on the same chart? If not, would it be clearer to do something like `legend = numericLegend || categoryLengend`

Reply:
Technically yes, if a variable contains both numerical and non-numerical values.

Categorical legends are automatically used for `string` values, numerical legends for `number` values. (This is after they are parsed, since in the database it's all `VARCHAR`)